paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/clone/CloneHiveUtils.java

```diff
@@ -0,0 +1,415 @@
+                                    schema.primaryKeys(),
+                                    options,
+                                    schema.comment());
+                    targetCatalog.createTable(tuple.f1, schema, false);
+                    FileStoreTable table = (FileStoreTable) targetCatalog.getTable(tuple.f1);
+                    List<HivePartitionFiles> allPartitions =
+                            HiveMigrateUtils.listFiles(
+                                    hiveCatalog,
+                                    tuple.f0,
+                                    table.schema().logicalPartitionType(),
+                                    table.coreOptions().partitionDefaultName());
+                    List<CloneFilesInfo> cloneFilesInfos = new ArrayList<>();
+                    for (HivePartitionFiles partitionFiles : allPartitions) {
+                        cloneFilesInfos.add(CloneFilesInfo.fromHive(tuple.f1, partitionFiles, 0));
+                    }
+                    collector.collect(cloneFilesInfos);
+                }
+            }
+        };
+    }
+
+    public static ProcessFunction<List<CloneFilesInfo>, Void> copyAndCommitFunction(
+            Map<String, String> sourceCatalogConfig, Map<String, String> targetCatalogConfig) {
+        return new ProcessFunction<List<CloneFilesInfo>, Void>() {
+
+            @Override
+            public void processElement(
+                    List<CloneFilesInfo> cloneFilesInfos,
+                    ProcessFunction<List<CloneFilesInfo>, Void>.Context context,
+                    Collector<Void> collector)
+                    throws Exception {
+                try (Catalog targetCatalog =
+                        FlinkCatalogFactory.createPaimonCatalog(
+                                Options.fromMap(targetCatalogConfig))) {
+
+                    // source FileIO
+                    CatalogContext sourceContext =
+                            CatalogContext.create(Options.fromMap(sourceCatalogConfig));
+                    String warehouse = checkNotNull(sourceContext.options().get(WAREHOUSE));
+                    FileIO sourceFileIO = FileIO.get(new Path(warehouse), sourceContext);
+
+                    // group by table
+                    Map<Identifier, List<CloneFilesInfo>> groupedFiles = new HashMap<>();
+                    for (CloneFilesInfo files : cloneFilesInfos) {
+                        groupedFiles
+                                .computeIfAbsent(files.identifier(), k -> new ArrayList<>())
+                                .add(files);
+                    }
+
+                    for (Map.Entry<Identifier, List<CloneFilesInfo>> entry :
+                            groupedFiles.entrySet()) {
+                        FileStoreTable targetTable =
+                                (FileStoreTable) targetCatalog.getTable(entry.getKey());
+                        commit(entry.getValue(), sourceFileIO, targetTable);
```

Comment:
Commit should be done in a committer node for one table.

Reply:
1. In `CloneHiveAction`,  `files` are partitioned by table and forward to `copyAndCommitFunction`;
2. In `copyAndCommitFunction`,  `CloneFilesInfo` are grouped by table. 

So I think I have commit metas for one table once?